app/lib/proxy_api/available_proxy.rb

```diff
@@ -0,0 +1,28 @@
+module ::ProxyAPI
+  class AvailableProxy < ::ProxyAPI::Resource
```

Comment:
this shouldn't be API class but some service object, it should ideally just wrap `ProxyAPI::Features` which already has public method to get all [features](https://github.com/theforeman/foreman/blob/develop/lib/proxy_api/features.rb#L8).